base/dsfs/dataset.go

```diff
@@ -220,48 +221,69 @@ func DerefDatasetCommit(ctx context.Context, store cafs.Filestore, ds *dataset.D
 	return nil
 }
 
-// CreateDataset places a new dataset in the store. Admittedly, this isn't a simple process.
-// Store is where we're going to
-// Dataset to be saved
-// Pin the dataset if the underlying store supports the pinning interface
-// All streaming files (Body, Transform Script, Viz Script) Must be Resolved before calling if data their data is to be saved
-func CreateDataset(ctx context.Context, store cafs.Filestore, ds, dsPrev *dataset.Dataset, pk crypto.PrivKey, pin, forceIfNoChanges, shouldRender bool) (path string, err error) {
+// SaveSwitches represents options for saving a dataset
+type SaveSwitches struct {
+	// Replace is whether the save is a full replacement or a set of patches to previous
+	Replace bool
+	// DryRun is whether the save should not be written to the refstore for real
+	DryRun bool
+	// Pin is whether the dataset should be pinned
+	Pin bool
+	// ConvertFormatToPrev is whether the body should be converted to match the previous format
+	ConvertFormatToPrev bool
+	// ForceIfNoChanges is whether the save should be forced even if no changes are detected
+	ForceIfNoChanges bool
+	// ShouldRender is deprecated, controls whether viz should be rendered
+	ShouldRender bool
+	// NewName is whether a new dataset should be created, guaranteeing there's no previous version
+	NewName bool
+	// FileHint is a hint for what file is used for creating this dataset
+	FileHint string
```

Comment:
string isn't _exactly_ a switch, but by using the name `Switch` we avoid colliding with the func-based `Options` convention we use elsewhere.